services/ftp/cmd.go

```diff
@@ -970,13 +1019,19 @@ func (cmd commandStor) Execute(conn *Conn, param string) {
 		conn.appendData = false
 	}()
 
-	bytes, err := conn.driver.PutFile(param, conn.dataConn, conn.appendData)
-	if err == nil {
-		msg := "OK, received " + strconv.Itoa(int(bytes)) + " bytes"
-		conn.writeMessage(226, msg)
-	} else {
+	if conn.user == "anonymous" {
+		conn.writeMessage(450, "Restricted: Anonymous user can't upload files")
+		return
+	}
+
+	nbytes, err := conn.driver.PutFile(param, conn.dataConn, conn.appendData)
+	if err != nil {
 		conn.writeMessage(450, fmt.Sprintln("error during transfer:", err))
+		return
 	}
+
+	msg := "OK, received " + strconv.Itoa(int(nbytes)) + " bytes"
```

Comment:
Could you use `fmt.Sprintf("OK, received %d bytes", " strconv.Itoa(int(nbytes)))` here?

Reply:
also fixed in PR213